### PR TITLE
feat: adding package command

### PR DESCRIPTION
**Pull request by erickzanardo**
Adds a `shorebird_tools package` command which will be used to make the package archives.

For now, it only includes the patch itself.

### CODE REVIEW

File: packages/shorebird_tools/lib/src/commands/package_command.dart [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  /// {@macro package_command}
+  PackageCommand({
+    required Logger logger,
+  }) : _logger = logger {
+    argParser
+      ..addOption(
+        'patch',
+        abbr: 'p',
+        mandatory: true,
+        help: 'The patch path to package',
```
Comment (bryanoltman) [inline packages/shorebird_tools/lib/src/commands/package_command.dart]:
```suggestion
        help: 'The path to the patch artifact which will be packaged',
```

File: packages/shorebird_tools/lib/src/commands/package_command.dart [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      )
+      ..addOption(
+        'output',
+        abbr: 'o',
+        mandatory: true,
+        help: 'The path to the output archive',
```
Comment (bryanoltman) [inline packages/shorebird_tools/lib/src/commands/package_command.dart]:
```suggestion
        help: 'Where to write the packaged patch archive',
```

File: packages/shorebird_tools/lib/src/commands/package_command.dart [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      );
+  }
+
+  @override
+  String get description => 'Packages a generated patch into an archive';
```
Comment (bryanoltman) [inline packages/shorebird_tools/lib/src/commands/package_command.dart]:
```suggestion
  String get description => 'Packages a patch artifact into an archive';
```

**Review by eseidel — APPROVED** [review]

File: packages/shorebird_tools/lib/src/commands/package_command.dart [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    final archiveFile = ArchiveFile(
+      p.basename(patchFilePath),
+      bytes.length,
+      bytes,
+    );
+
+    ZipEncoder()
+      ..startEncode(OutputFileStream(outFile.absolute.path))
+      ..addFile(archiveFile)
+      ..endEncode();
```
Comment (bryanoltman) [inline packages/shorebird_tools/lib/src/commands/package_command.dart:64]: Maybe doesn't need to be in this PR, but in the CLI, we've generally put these in an isolate so as not to block progress logs etc.

File: packages/shorebird_tools/test/src/commands/package_command_test.dart [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+      await extractFileToDisk(patchFile.path, extractedFolder.path);
+
+      // Making sure it was correctly archived and can be extracted
+      final extractedFile = File(p.join(extractedFolder.path, 'patch.txt'));
+      expect(extractedFile.existsSync(), isTrue);
+      expect(extractedFile.readAsStringSync(), 'banana');
+    });
+
+    group('wrong usage', () {
+      test('when missing option for -p', () async {
```
Comment (bryanoltman) [inline packages/shorebird_tools/test/src/commands/package_command_test.dart]: Test names should generally describe what happens. "when missing option for -p" is a good group name. The test name is probably "logs error and exits with usage code"

File: packages/shorebird_tools/test/src/commands/package_command_test.dart [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+Usage: shorebird_tools package [arguments]
+-h, --help                  Print this usage information.
+-p, --patch (mandatory)     The patch path to package
+-o, --output (mandatory)    The path to the output archive
+
+Run "shorebird_tools help" to see global options.'''),
+        ).called(1);
+      });
+
+      test('when missing option for -o', () async {
```
Comment (bryanoltman) [inline packages/shorebird_tools/test/src/commands/package_command_test.dart]: Same here

**Review by bryanoltman — APPROVED** [review]
LGTM with some minor comments

**Review by eseidel — APPROVED** [review]

**Pull request merged**